core/src/main/scala/besom/util/NonEmptyString.scala

```diff
@@ -29,6 +29,31 @@ object NonEmptyString:
         */
       inline def asString: String = nes
 
+  object OutputOps:
+    import besom.internal.{Output, Context}
+    import besom.util.interpolator.*
+    inline def fromStringOutput(inline s: Output[String]): Output[NonEmptyString] = ${ fromStringOutputImpl('s) }
+
+    private def fromStringOutputImpl(expr: Expr[Output[String]])(using quotes: Quotes): Expr[Output[NonEmptyString]] =
```

Comment:
wow, I didn't know you can poke inside interpolators with macros like that, impressive